HPXMLtoOpenStudio/resources/hvac.rb

```diff
@@ -1,98 +1,135 @@
                                                  control_zone, hvac_map)
-
     hvac_map[cooling_system.id] = [] unless cooling_system.nil?
     hvac_map[heating_system.id] = [] unless heating_system.nil?
-    if heating_system.nil?
-      obj_name = Constants.ObjectNameCentralAirConditioner
-    elsif cooling_system.nil?
-      obj_name = Constants.ObjectNameFurnace
-    else
-      obj_name = Constants.ObjectNameCentralAirConditionerAndFurnace
-    end
 
-    if not heating_system.nil?
-      htg_ap = heating_system.additional_properties
-    end
-    if not cooling_system.nil?
-      clg_ap = cooling_system.additional_properties
-    end
-
-    # Cooling Coil
+    is_heatpump = false
     if not cooling_system.nil?
+      if cooling_system.is_a? HPXML::HeatPump
+        is_heatpump = true
+        if cooling_system.heat_pump_type == HPXML::HVACTypeHeatPumpAirToAir
+          obj_name = Constants.ObjectNameAirSourceHeatPump
+        elsif cooling_system.heat_pump_type == HPXML::HVACTypeHeatPumpMiniSplit
+          obj_name = Constants.ObjectNameMiniSplitHeatPump
+        elsif cooling_system.heat_pump_type == HPXML::HVACTypeHeatPumpPTHP
+          obj_name = Constants.ObjectNamePTHP
+          fan_watts_per_cfm = 0.0
+        else
+          fail "Unexpected heat pump type: #{cooling_system.heat_pump_type}."
+        end
+      else
+        if cooling_system.cooling_system_type == HPXML::HVACTypeCentralAirConditioner
+          if heating_system.nil?
+            obj_name = Constants.ObjectNameCentralAirConditioner
+          else
+            obj_name = Constants.ObjectNameCentralAirConditionerAndFurnace
+            # error checking for fan power
+            if (cooling_system.fan_watts_per_cfm.to_f != heating_system.fan_watts_per_cfm.to_f)
+              fail "Fan powers for heating system '#{heating_system.id}' and cooling system '#{cooling_system.id}' are attached to a single distribution system and therefore must be the same."
+            end
+          end
+        elsif cooling_system.cooling_system_type == HPXML::HVACTypeRoomAirConditioner
+          obj_name = Constants.ObjectNameRoomAirConditioner
+          fan_watts_per_cfm = 0.0
+        elsif cooling_system.cooling_system_type == HPXML::HVACTypeMiniSplitAirConditioner
+          obj_name = Constants.ObjectNameMiniSplitAirConditioner
+        elsif cooling_system.cooling_system_type == HPXML::HVACTypePTAC
+          obj_name = Constants.ObjectNamePTAC
+          fan_watts_per_cfm = 0.0
+        else
+          fail "Unexpected cooling system type: #{cooling_system.cooling_system_type}."
+        end
+      end
+      # Cooling Coil
       clg_coil = create_dx_cooling_coil(model, obj_name, cooling_system)
       hvac_map[cooling_system.id] << clg_coil
+
+      clg_ap = cooling_system.additional_properties
+      num_speeds = clg_ap.num_speeds
+      clg_cfm = cooling_system.cooling_airflow_cfm
+    else
+      obj_name = Constants.ObjectNameFurnace
+      num_speeds = 1
```

Comment:
Wrap this inside an `if heating_system.heating_system_type == HPXML::HVACTypeFurnace` just to be safe.